packages/calypso-e2e/src/lib/pages/editor-page.ts

```diff
@@ -563,39 +540,65 @@ export class EditorPage {
 	//#region Publish, Draft & Schedule
 
 	/**
-	 * Publishes the post or page.
+	 * Publishes the post or page and returns the resulting URL.
+	 *
+	 * If the optional parameter `visit` parameter is specified, the page is navigated
+	 * to the published article.
 	 *
 	 * @param {boolean} visit Whether to then visit the page.
 	 * @returns {URL} Published article's URL.
 	 */
 	async publish( { visit = false }: { visit?: boolean } = {} ): Promise< URL > {
-		// Click on the main publish action button on the toolbar.
-		await this.editorToolbarComponent.clickPublish();
+		const publishButtonText = await this.editorToolbarComponent.getPublishButtonText();
+		const actionsArray = [];
+
+		// Playwright does not have a way to differentiate
+		// between GET and POST requests. However, a new post
+		// has a post number following the article type.
+		// By matching on the regex we can filter out
+		// GET requests to the endpoint, focusing only on POST requests.
+		if ( this.target === 'atomic' ) {
+			actionsArray.push( this.page.waitForResponse( /v2\/(posts|pages)\/[\d]+/ ) );
+		} else {
+			actionsArray.push( this.page.waitForResponse( /sites\/[\d]+\/(posts|pages)\/[\d]+.*/ ) );
+		}
 
-		if ( await this.editorPublishPanelComponent.panelIsOpen() ) {
-			// Invoke the second stage of the publish step which handles the
-			// publish checklist panel if it is present.
-			await this.editorPublishPanelComponent.publish();
```

Comment:
The refactor of `publish` method is little bit of a scope creep that was prompted by the removal of the `getEditorHandle` method entirely. 

As the original `getPublishedURLFromToast` method used the now-removed `getEditorHandle` method, I figured I'll take this opportunity to refactor how the published article's URL is obtained. My primary reasoning is that using UI elements only is flaky.

With that said, if this should be in a separate PR, I am all for it.